gcc/rust/resolve/rust-ast-resolve-type.cc

```diff
@@ -30,11 +30,28 @@ ResolveTypeToCanonicalPath::canonicalize_generic_args (AST::GenericArgs &args)
   std::string buf;
 
   size_t i = 0;
-  size_t total = args.get_type_args ().size ();
+  size_t total = args.get_generic_args ().size ();
 
-  for (auto &ty_arg : args.get_type_args ())
+  for (auto &arg : args.get_generic_args ())
     {
-      buf += ty_arg->as_string ();
+      // FIXME: Arthur: Is that function used solely for pretty-printing/dumping
```

Comment:
This is changed in the refactor. There is a kind of similar function but its used to generate the canonical path to something like:

impl<T> Foo<T> { fn test(self) {} }

Where the path is expected to be crate::<impl Foo<T>>::test iirc